components/table/index.jsx

```diff
@@ -52,20 +56,26 @@ class InfiniteTable extends React.Component {
     this.setState(state => ({
       dataRequestCount: state.dataRequestCount + 1,
     }))
-
     const newState = {}
 
-    const data = await fetchData(pageNumber, searchTerm, columnOrder)
-    if (data.length === 0) newState.isLastPageLoaded = true
-    if (data.length === 0 && pageNumber === 0) newState.isEmpty = true
-    if (pageNumber === 0) newState.isFirstPageLoaded = true
+    return fetchData(pageNumber, searchTerm, columnOrder).then(page => {
+      const { data } = page
+      // eslint-disable-next-line no-shadow
+      const { searchTerm } = this.state
```

Comment:
What the problem is here, why could not you avoid `eslint-disable`?

By the way, state is asynchronous and checking request count from there is not really good idea.

Reply:
RequestCount is always changed via function

```
   this.setState(state => ({
      dataRequestCount: state.dataRequestCount + 1,	
    }))
```


If you meant `searchTerm` I think it's fine and it worked for me. I don't know how could I solve it differently.

Reply:
We can store request controller in store and for each new request call controller.abort() and create new request. Find out more about [Abortable fetch](https://developers.google.com/web/updates/2017/09/abortable-fetch) at Goole Developers blog.

To achieve this you will need to change API request signature. It would be good to do it anyway.

Reply:
I was thinking about it too but it's not that easy. This was a much simple solution.

Reply:
Simple is not always good. Debounce and stuff is more complex to understand, increases user's network load and may cause problems.

We should do timeout at the interface part to prevent sending requests for each letter. However, store should sent request when it's asked for. Could you do that ignoring aborting part?